src/bernsteinyang.rs

```diff
@@ -0,0 +1,351 @@
+use core::cmp::PartialEq;
+use std::ops::{Add, Mul, Sub, Neg};
+
+/// Big signed (B * L)-bit integer type, whose variables store
+/// numbers in the two's complement code as arrays of B-bit chunks.
+/// The ordering of the chunks in these arrays is little-endian.
+/// The arithmetic operations for this type are wrapping ones.
+#[derive(Clone)]
+struct ChunkInt<const B:usize, const L:usize>(pub [u64; L]);
```

Comment:
It seems that this term is rarely used. On the other hand, "saturated integer" is usually used to describe the type, for which the saturation arithmetic is implemented: https://en.wikipedia.org/wiki/Saturation_arithmetic A least, my first thought was about the saturation arithmetic, when I saw this term somewhere before. Thus, I will leave this comment "as is" to avoid the ambiguation.